reconcile/openshift_saas_deploy_trigger_base.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+from reconcile.jenkins_job_builder import get_openshift_saas_deploy_job_name
+
+
+def trigger(options):
+    """Trigger a deployment according to the specified pipelines provider
+
+    Args:
+        options (dict): A dictionary containing:
+            dry_run (bool): Is this a dry run
+            spec (dict): A trigger spec as created by saasherder
+            jenkins_map (dict): Instance names with JenkinsApi instances
+            already_triggered (list): A list of already triggered deployments
+            settings (dict): App-interface settings
+            state_update_method (function): A method to call to update state
+
+    Returns:
+        bool: True if there was an error, False otherwise
+    """
+    dry_run = options['dry_run']
+    spec = options['spec']
+    jenkins_map = options['jenkins_map']
+    already_triggered = options['already_triggered']
+    settings = options['settings']
+    state_update_method = options['state_update_method']
+
+    saas_file_name = spec['saas_file_name']
+    env_name = spec['env_name']
+    pipelines_provider = spec['pipelines_provider']
+    provider_name = pipelines_provider['provider']
```

Comment:
I was trying to avoid a long list of arguments for the function. I actually did what you said in the first place, but this class mostly uses these `options`, so i went for consistency instead. wdyt?

Reply:
I prefer the long list of arguments here, if that's what we have to have.